network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +340,26 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `MAX_FULLNODE_CONNECTIONS`
+        // including in flight dials.
+        let to_connect_size = if let RoleType::FullNode = self.network_context.role() {
+            min(
+                to_connect.len(),
+                MAX_FULLNODE_CONNECTIONS - self.connected.len() - self.dial_queue.len(),
+            )
+        } else {
+            to_connect.len()
+        };
```

Comment:
yeah, not entirely surprised on the size not decreasing :/ but at least now it is more syntactically correct if else is better than a single match else